katdal/datasources.py

```diff
@@ -207,6 +207,32 @@ def view_capture_stream(telstate, capture_block_id=None, stream_name=None):
     return telstate
 
 
+def _shorten_key(telstate, key):
+    """Shorten telstate key by subtracting the first prefix that fits.
+
+    Parameters
+    ----------
+    telstate : :class:`katsdptelstate.TelescopeState` object
+        Telescope state
+    key : string
+        Telescope state key
+
+    Returns
+    -------
+    short_key : string or None
+        Suffix of `key` after subtracting first matching prefix, or None if
+        `key` does not start with any of the prefixes
+
+    """
+    for prefix in telstate.prefixes:
+        if not prefix:
+            return key
+        head, sep, tail = key.partition(prefix)
```

Comment:
I had not come across this function before. But it seems slightly roundabout - I would have written the whole loop content as:
```python
if key.startswith(prefix):
    return key[len(prefix):]
```

Reply:
Good point.